src/crypto/checksum.rs

```diff
@@ -80,16 +79,26 @@ impl Hasher for SimpleChecksum {
     }
 }
 
-/// SHA1 checksum, first 20 octets.
+/// SHA1 checksum, using sha1_checked, first 20 octets.
+///
+/// Fails with `Error::HashCollision` if a SHA1 collision was detected.
 #[inline]
-pub fn calculate_sha1<I, T>(data: I) -> Vec<u8>
+pub fn calculate_sha1<I, T>(data: I) -> Result<Vec<u8>>
```

Comment:
return an array, and only convert into vec when needed at call site?

Reply:
Right, sounds good.

(Initially I thought it's better to keep the `Vec<u8>` return type just to avoid gratuitous API breakage. But we're now changing the type to `Result` anyway.)